core/src/main/java/org/mule/routing/AsynchronousUntilSuccessfulProcessingStrategy.java

```diff
@@ -215,8 +217,11 @@ public static Serializable buildQueueKey(final MuleEvent muleEvent)
         // the key is built in way to prevent UntilSuccessful workers across a
         // cluster to compete for the same
         // events over a shared object store
-        String key = muleEvent.getFlowConstruct() + "-" + muleEvent.getMuleContext().getClusterId() + "-"
-                     + muleEvent.getId();
+        // it also adds a random trailer to support events which have been
+        // splitted and thus have the same id. Random number was chosen over
+        // UUID for performance reasons
+        String key = String.format("%s-%s-%s-%d", muleEvent.getFlowConstruct(), muleEvent.getMuleContext().getClusterId(), muleEvent.getId(), random.nextInt());
```

Comment:
I'm not sure if the key can be random. @pablolagreca can you confirm if this is OK?